docs/design-docs/02-custom-root-reconciler-clusterrole.md

```diff
@@ -18,65 +24,130 @@ This proposal introduces a mechanism to customize this behavior.
 If one wants to allow deploying resources from a source where one doesn't have full
 control of the contents of the upstream config source, and want to limit what another
 team or third party can accomplish in the cluster, one can imagine limiting what
-permissions are granted to the `ServiceAccount` running the reconciler.
+permissions are granted to the `ServiceAccount` running the `RootSync` reconciler.
 
 However, as Config Sync is currently granting `cluster-admin`, any custom role bindings
-are effectively ignored; the reconciler will, due to the binding added by Config Sync,
+are effectively ignored; the `RootSync` reconciler will, due to the binding added by Config Sync,
 have access to do _anything_ regardless.
 
 To follow the [principle of least privilege], one should ensure the reconciler only has
 access to deploy the expected resources.
 
+Additionally, the `RepoSync` API currently requires for users to manage the lifecycle
+of `RoleBinding` objects themselves. The proposed API here is intended to also
+simplify the management of `RoleBinding`s for `RepoSync`s.
+
 See also [#935].
 
 [principle of least privilege]: https://en.wikipedia.org/wiki/Principle_of_least_privilege
 [#935]: https://github.com/GoogleContainerTools/kpt-config-sync/issues/935
 
 ## Design Overview
 
-By providing the name of a user-defined `ClusterRole` for a `RootSync`, a user can
+### `RootSync`s
+
+By providing the name of a list user-defined `RoleRef`s for a `RootSync`, a user can
 override which role Config Sync binds to. This configuration is exposed as a new field
 on `spec.overrides` for a `RootSync`:
 
 ```yaml
+kind: RootSync
+metadata:
+  name: my-root-sync
+  namespace: config-management-system
 spec:
   overrides:
-    clusterRole:
-      name: my-custom-role
+    roleRefs:
+    - kind: ClusterRole # Creates a ClusterRoleBinding bound to ClusterRole my-cluster-role
```

Comment:
nit: the CRB isn't the object being bound, it's the subject doing the binding. The CRB binds the CR to the KSA.

Reply:
rephrased